### PR TITLE
LRQA-62432 Add portlet to test Alert Toast Message

### PR DESCRIPTION
**Pull request by zhangyan0701**
https://issues.liferay.com/browse/LRQA-62432

The portlet is to be used for testing Clay Toast Alert in Poshi tests.

This is how it looks like after deploying to a page:
test portlet

Please take a look if this is ok for testing. Thanks!

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
Closed, will resend after resolving the SF failure.

**Comment by john-co** [comment]
I reopened this as a draft so it can still be visible for review from others.

File: modules/apps/frontend-js/frontend-js-test/frontend-js-test-clay-toast-alert-sample-web/src/main/java/com/liferay/frontend/js/test/clay/toast/alert/sample/web/internal/portlet/FrontendJSTestClayToastAlertSampleWebPortlet.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+package com.liferay.frontend.js.test.clay.toast.alert.sample.web.internal.portlet;
+
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet;
+
+import javax.portlet.Portlet;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Summer Zhang
```
Comment (john-co) [inline modules/apps/frontend-js/frontend-js-test/frontend-js-test-clay-toast-alert-sample-web/src/main/java/com/liferay/frontend/js/test/clay/toast/alert/sample/web/internal/portlet/FrontendJSTestClayToastAlertSampleWebPortlet.java:24]: :sunglasses:

**Review by john-co — COMMENTED** [review]
:clap: :clap: :star_struck: this looks awesome @zhangyan0701 . Nice touch with the alert stripe :ok_hand: 

I'm still thinking about how to best store our test artifacts in the long run...I think this would be more related to `apps/frontend-taglib` directory so here's my thoughts:

**Option A:**
- Place test portlets in app directory: `frontend-taglib/frontend-js-test-clay-toast-alert-sample-web`
but if we follow this pattern, when we eventually build more test portlets, I feel it would bloat up frontend-taglib.

**Option B:**
- Place test portlets in app test subdirectory: `frontend-taglib/frontend-taglib-test/frontend-js-test-clay-toast-alert-sample-web`

**Option C:**
- Place test portlets in dependencies folder following pattern of QA portal-web and other apps. ie: `frontend-taglib/frontend-taglib-test/src/testFunctional/dependencies`

I'm leaning towards Option B to reduce adding tests in main app directory. For Option C maybe not add uncompiled code, better if it was a .jar file or something.

Thoughts on how we want to organize our test portlets @jbalsas @wincent?

**Comment by jbalsas** [comment]
> I'm still thinking about how to best store our test artifacts in the long run...I think this would be more related to apps/frontend-taglib directory so here's my thoughts:

One thing with tests is that the closest they are to the source the easiest it is to keep them working. To that end, I feel like co-locating them directly within `frontend-taglib-clay` could be desirable as long as we could avoid the undesired effects of polluting the production bundle with test portlets. This maybe can be done by excluding `test` folders in the `bnd.bnd`. Language properties would also be a cause of concern if we did that, but maybe we could also figure out a way around that.

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
Fix the SF first.

**Comment by john-co** [comment]
Let's move this portlet in `frontend-taglib/frontend-taglib-clay/`. I also want to rename the portlet to something more easier to understand the purpose, like `test-app-alert-toast`.

We can later follow up in another ticket with seeing how possible it is to migrate to `frontend-taglib/frontend-taglib-clay/test/alert_toast`.

**Comment by john-co** [comment]
follow up ticket: https://issues.liferay.com/browse/LRQA-66231

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
Hi @john-co 

The module name has to be start with 'frontend-taglib-clay-test', otherwise the SF will fail. So I remove the module to `frontend-taglib//frontend-taglib-clay-test/frontend-taglib-clay-test-alert-toast`.

Is this name ok?

**Comment by john-co** [comment]
ic, I think the name is fine for now.

Just make sure to add a .lfrbuild-releng-ignore file so it won't be included in release.

Looks good to forward after that.

**Review by john-co — APPROVED** [review]

**Comment by zhangyan0701** [comment]
ci:forward

**Pull request closed without merging**